docs/quick-starts/rug-typescript.md

```diff
@@ -0,0 +1,202 @@
+
+Once you have these two installed on your machine you should be able to successfully execute `npm install` in your `.atomist` directory to see the following:
+
+```
+$ npm install
+@russmiles/scattered-rugs@0.8.0 ...scattered-rugs/.atomist
+└── @atomist/rug@0.10.0
+```
+
+This will create a `node_modules` directory which can be ignored, using something like `.gitignore` if you're using [git](https://git-scm.com/), and not checked into source control as it is only needed for local development.
+
+### Writing your first TypeScript Rug editor
+
+Now you are set up for working with Rug in TypeScript, let's write a simple editor. As usual, we start by constructing a test for our future editor and we can do this using the convenience of a Rug DSL BDD test:
+
+```
+scenario SimpleSampleEditor should just add a file called "README.md" to the target project
+
+given
+  Empty
+
+when
+  SimpleSampleEditor
+
+then
+  fileExists "README.md"
+    and fileContains "README.md" "Hello, Rug TypeScript World!"
+```
+
+Drop this test into a file called `.atomist/tests/MyFirstEditor.rt` and then execute `rug test` using the Rug CLI (link) and you should see something like the following:
+
+```
+$ rug test
+Resolving dependencies for russmiles:scattered-rugs:0.1.0 ← local completed
+Loading russmiles:scattered-rugs:0.1.0 ← local into runtime completed
+Running test scenarios in russmiles:scattered-rugs:0.1.0 ← local completed
+
+→ Failed Scenarios
+  SimpleSampleEditor should just add a file called "README.md" to the target project (1 of 1 assertions failed)
+    Failed Assertions
+       Scenario 'SimpleSampleEditor should just add a file called "README.md" to the target project' tests editor 'SimpleSampleEditor', which was not found.
+Known operations are []
+```
+
+With our test *happily* failing you can now write the following Rug TypeScript editor to meet the assertions of the test. Create a file called `SimpleSampleEditor.ts` in the `.atomist/editors` directory that contains the following:
+
+```
+import { ProjectEditor } from "@atomist/rug/operations/ProjectEditor"
+import { Status, Result, Parameter } from "@atomist/rug/operations/RugOperation"
+import { Project, Pair, File } from '@atomist/rug/model/Core'
+import { PathExpression, PathExpressionEngine, TreeNode, Match } from '@atomist/rug/tree/PathExpression'
+
+
+let params: Parameter[] = [
+    {
+        name: "description",
+        displayName: "Description",
+        description: "Text to be added to the README.md",
+        validInput: "Any valid text",
+        minLength: 1,
+        pattern: "@any",
+    }
+]
+
+interface Parameters {
+    description: string
+}
+
+export let editor: ProjectEditor = {
+    tags: ["simple"],
+    name: "SimpleSampleEditor",
+    description: "A simple sample TypeScript editor",
+    parameters: params,
+    edit(project: Project, p: Parameters): Result {
```

Comment:
I think the destructuring form might be better here:

edit(project: Project, {description } : {description: string}) {

}

And sack off the Parameters interface.